### ISSUE TITLE
has no initializer and is not definitely assigned in the constructor

### ISSUE THREAD

**Issue by ajjack50n** [sequelize/sequelize-typescript#1332]
# Issue
typescript complains :no initializer and is not definitely assigned in the constructor    

## Versions

- sequelize: 6.21
- sequelize-typescript: 2.1.3
- typescript: 4.7.2

## Issue type

- [x ] bug report
- [ ] feature request

## Actual behavior
setting up model as below and typescript complains that no initializer assigned...
I realise I can turn this warning off but docs do not say I should need to do this, It feels wrong to have to do so?

Also I get warning:

(sequelize) Warning: Model "User" is declaring public class fields for attribute(s): "id", "name", "surname", "email", "image", "emailVerified", "password", "pronoun".
These class fields are shadowing Sequelize's attribute getters & setters.

Again, I am wondering if this is expected behavior but if so, perhaps this should be documented,

The model works other than the annoyance of above.

```
@Table
class User extends Model<User> {
    @PrimaryKey @Column({ type: DataType.CHAR(36) })
    id: string

    @Column
    name: string

    @Column
    surname: string

    @Column
    email: string

    @Column
    image: string

    @Column
    emailVerified: boolean

    @Column
    password: string

    @Column
    pronoun: string

}

```

## Expected behavior

I would not expect to get warning as noted above, or else, at least this to be documented as expected.

## Steps to reproduce
create model as described above...

## Related code

```ts
insert short code snippets here
```

**Comment by ajjack50n** [comment]
So I think I have answered my own question by looking at the sample project linked to the documentation. It seems that that we should use def and optional assignment declarations.  This makes sense but I feel the documents should make this clear for noobs like me. 

Also I still have the question around the Getter / Setter shadowing. I think this is probably to be expected but again, it would be useful if the documentation could call this out.

**Comment by ephys** [comment]
The warning gives you a link to a page of the documentation that explains what it is and what should be done

**Comment by ajjack50n** [comment]
hey. .thanks for response... you are correct the link provided leads me to conclude and expect the issue but this could be more explicit in the documentation. 

You say 'what should be done' but unless I am missing something from that documentation, I cant see what it says I can do? My understanding is that this library requires me to decorate public fields with meta-data annotations? I have tried using declare instead - as hinted at by the link but that does not seem to work well  - but that could be me messing up Babel config.
That said I note the linked project does not use declare either.

**Comment by ajjack50n** [comment]
I think I have  stumbled across an exiting issue. 

https://github.com/sequelize/sequelize-typescript/issues/1236#issue-1180396878

**Comment by paul-uz** [comment]
@ajjack50n what is the solution to this? I'm getting the same thing with a super basic model

```
import { Column, Model, PrimaryKey, Table } from 'sequelize-typescript';

@Table
export class Match extends Model {
  @Column
  @PrimaryKey
  matchID!: string;

  @Column
  competitionID: number; // Property 'competitionID' has no initializer and is not definitely assigned in the constructor.ts(2564)
}
```

**Comment by ajjack50n** [comment]
It's a while since I was playing with this. Look at my comment above, I think I worked around it using in tsc using declare statement.

**Comment by paul-uz** [comment]
> It's a while since I was playing with this. Look at my comment above, and the referenced issue, I think I worked around it using in tsc using declare statement.

Turns out @Column had to be after @PrimaryKey 🙄
